ouroboros-consensus/src/Ouroboros/Consensus/Node/Run.hs

```diff
@@ -73,6 +76,16 @@ class ( ConvertRawHash blk
       , SerialiseResult       blk (Query blk)
       ) => SerialiseNodeToClientConstraints blk
 
+-- | A bespoke 3-valued boolean: yes, no, or maybe.
+data WhetherSelfIssued =
+    -- | A block we produced ourself
+    IsSelfIssued
+    -- | A block produced by another node
+  | IsNotSelfIssued
+    -- | We do not have enough information to decide
+  | MaybeSelfIssued
```

Comment:
Rename `Maybe` to `Unknown`; avoids some accidental connotations. Implies either `likely` or `we could figure out with some more effort` -- neither of which is what we want here.